src/context/WorkoutContext.tsx

```diff
@@ -0,0 +1,249 @@
+                })),
+            );
+        }
+    };
+
+    // Resume a paused timer
+    const resumeTimer = () => {
+        if (currentTimerIndex !== null && timers[currentTimerIndex]?.state === 'paused') {
+            setTimers(prevTimers =>
+                prevTimers.map((timer, index) => ({
+                    ...timer,
+                    state: index === currentTimerIndex ? 'running' : timer.state,
+                })),
+            );
+        }
+    };
+
+    // Pause the currently active timer
+    const pauseTimer = () => {
+        if (currentTimerIndex !== null) {
+            setTimers(prevTimers =>
+                prevTimers.map((timer, index) => ({
+                    ...timer,
+                    state: index === currentTimerIndex ? 'paused' : timer.state,
+                })),
+            );
+        }
+    };
+
+    // Move to the next timer
+    const nextTimer = (skip = false) => {
+        if (currentTimerIndex !== null) {
+            const nextIndex = currentTimerIndex + 1;
+
+            setTimers(prevTimers =>
+                prevTimers.map((timer, index) => {
+                    // Handle the current timer
+                    if (index === currentTimerIndex) {
+                        return {
+                            ...timer,
+                            state: 'completed', // Mark current timer as completed
+                            skipped: skip ? true : timer.skipped, // Mark as skipped if applicable
+                        };
+                    }
+
+                    // Handle the next timer
+                    if (index === nextIndex) {
+                        return {
+                            ...timer,
+                            state: 'running', // Start the next timer
+                        };
+                    }
+
+                    // Return other timers unchanged
+                    return timer;
+                }),
+            );
+
+            if (nextIndex < timers.length) {
+                setCurrentTimerIndex(nextIndex); // Update to the next timer
+                setElapsedTime(0); // Reset elapsed time for the new timer
+            } else {
+                // No more timers, end the workout
+                setCurrentTimerIndex(null);
+            }
+        }
+    };
+
+    // Reset the workout
+    const resetWorkout = () => {
+        setElapsedTime(0); // Reset elapsed time
+        setTotalElapsedTime(0); // Reset total elapsed time
+        setIsWorkoutEditable(true); // Re-enable editing after reset
+        setCurrentTimerIndex(null);
+        setTimers(prevTimers =>
+            prevTimers.map(timer => ({
+                ...timer,
+                state: 'not running',
+                // currentRound: 1,
+                skipped: false, // Reset skipped status
+            })),
+        );
+    };
+
+    // Calculate total workout time
+    useEffect(() => {
+        const totalWorkoutTime = timers.reduce((total, timer) => {
```

Comment:
A lot of logic here - you could break this calculation out into a helper function to make the Context logic a little easier to read.